### PR TITLE
fix: do not break with pending invitations + increase timeout

### PR DESCRIPTION
**Pull request by miquelbar**
* increase timeout to 30 seconds. I got `│ Error: Get "https://app.honeybadger.io/v2/teams": context deadline exceeded (Client.Timeout exceeded while awaiting headers)`
* Do not try to re-add a user when it has a pending invitation. Honeybadger team added `invitations` into `/v2/teams` API endpoint to prevent extra calls to the API in order to know if a user it's a member or it has a pending invitation.

### CODE REVIEW

File: cli/users.go [hunk 1]
```diff
@@ -106,6 +106,8 @@ func (hbc *HoneybadgerClient) DeleteUser(userID int, teamID int) error {
 
 // GetUserFromTeams - Get User information from Teams
 func (hbc *HoneybadgerClient) GetUserFromTeams(userEmail string) (userTeams []HoneybadgerUser, err error) {
```
Comment (mpucholblasco) [inline cli/users.go:108]: I guess that you can simply the entire function by using something like:
```
func (hbc *HoneybadgerClient) GetUserFromTeams(userEmail string) (userTeams []HoneybadgerUser, err error) {
	teams, err := hbc.GetTeams()
	if err != nil {
		return userTeams, err
	}

	team_loop:
	for _, team := range teams {
		for _, user := range team.Users {
			if user.Email != userEmail {
				continue
			}
			user.TeamID = team.ID
			userTeams = append(userTeams, user)
			continue team_loop
		}

		for _, userInvitation := range team.Invitations {
			if userInvitation.Email != userEmail {
				continue
			}

			userTeams = append(
				userTeams,
				HoneybadgerUser{
					ID:        userInvitation.ID,
					Email:     userInvitation.Email,
					IsAdmin:   userInvitation.IsAdmin,
					TeamID:    team.ID,
					CreatedAt: userInvitation.CreatedAt,
				},
			)
			continue team_loop
		}
	}

	return userTeams, nil
}
```

**Review by mpucholblasco — APPROVED** [review]

File: cli/users.go (hunk 1 above)
Comment (miquelbar) [reply cli/users.go:108]: I don't get your point of view.

The map is used to only iterate if the user is not in member team. As a consequence, I need to check if the user is inside invitations array.

Comment (mpucholblasco) [reply cli/users.go:108]: Yes, but the `userEmail` is passed as parameter in the function, so imagine the following case:
* User `marcel.puchol@sequra.es` belongs to two teams (team `A` and team `B`) and has been invited in the team `C`
* In the first iteration, you will store `insertedUser["marcel.puchol@sequra.es"] = true`
* Then, once you will check the conditional `insertedUser["marcel.puchol@sequra.es"]`, it will be true, so I guess that you should check if the user is present in a concrete team. Being this the case and only being possible to be present in a team with a concrete email (it means, in the team `A`, the email "marcel.puchol@sequra.es" should only be present once, you can improve the loop by the previous proposal

Comment (miquelbar) [reply cli/users.go:108]: Yes, you're right in what you're saying. However, I would like to use the benefits of a `map` instead of `continue`. 

Also, I'll add these corner cases into unit test 😄

File: cli/users.go [hunk 2]
```diff
@@ -104,27 +104,48 @@ func (hbc *HoneybadgerClient) DeleteUser(userID int, teamID int) error {
...
+	insertedUser := make(map[string]bool)
+
 	teams, err := hbc.GetTeams()
 	if err != nil {
 		return userTeams, err
 	}
 
 	for _, team := range teams {
+		delete(insertedUser, userEmail) // Ensure that It is only valid fro the current team
```
Comment (mpucholblasco) [inline cli/users.go]: If this is the case, why not just creating a boolean here?
```
insertedUser := false
```
then, substitute `insertedUser[userEmail] = true` by `insertedUser = true` and finally in the check, only  check by the value of this boolean?

Sorry, but I don't see the benefit of using a map here. Could you explain me which are those?
and then in the conditional check for the

**Review by mpucholblasco — APPROVED** [review]

File: cli/users.go (hunk 2 above)
Comment (miquelbar) [reply cli/users.go]: Good point 😄

**Review by mpucholblasco — APPROVED** [review]
